src/ripple/app/ledger/impl/LedgerMaster.cpp

```diff
@@ -1037,6 +1039,53 @@ LedgerMaster::checkAccept(std::shared_ptr<Ledger const> const& ledger)
     app_.getFeeTrack().setRemoteFee(fee);
 
     tryAdvance();
+
+    if (ledger->seq() % 256 == 0)
+    {
+        // Check if the majority of validators run a higher version rippled
+        // software. If so print a warning.
+        //
+        // Once the HardenedValidations amendment is enabled, validators include
+        // their rippled software version in the validation messages of every
+        // (flag - 1) ledger. We wait for one ledger time before checking the
+        // version information to accumulate more validation messages.
+
+        auto const& parentHash = ledger->info().parentHash;
+        auto const parentLedger = getLedgerByHash(parentHash);
+        if (parentLedger &&
+            parentLedger->rules().enabled(featureHardenedValidations))
+        {
+            // To throttle the warning messages, instead of printing a warning
+            // every flag ledger, we print every week.
+            auto currentTime = app_.timeKeeper().now();
+            if (currentTime - upgradeWarningPrevTime_ >= weeks{1})
```

Comment:
I think that having the first warning come as soon as possible is important, but I'm not speaking ex cathedra.

Reply:
I also vote for having the first warning come sooner than several days on average. One thing I should probably do is to reduce the frequency of checking to 1024 ledgers instead of 256 ledgers. That means checking roughly every hour. 

As a small optimization, the code is changed to print every week after the first print, without checking if the majority of validators run a newer version (assuming most of them do not downgrade).